routing/payment_lifecycle.go

```diff
@@ -26,166 +28,445 @@ func (e errNoRoute) Error() string {
 		e.lastError)
 }
 
+// paymentFailure represents a terminal failure that can happen during the
+// payment lifecycle. We use it to record such an encountered error while we
+// wait for the remaining shards to finish.
+type paymentFailure struct {
+	failureCode channeldb.FailureReason
+	err         error
+}
+
 // paymentLifecycle holds all information about the current state of a payment
 // needed to resume if from any point.
 type paymentLifecycle struct {
-	router         *ChannelRouter
-	payment        *LightningPayment
-	paySession     PaymentSession
-	timeoutChan    <-chan time.Time
-	currentHeight  int32
-	finalCLTVDelta uint16
-	attempt        *channeldb.PaymentAttemptInfo
-	circuit        *sphinx.Circuit
-	lastError      error
+	router           *ChannelRouter
+	totalAmount      lnwire.MilliSatoshi
+	paymentHash      lntypes.Hash
+	paySession       PaymentSession
+	timeoutChan      <-chan time.Time
```

Comment:
This timeoutChan only seems to be in use for regular payments. SendToRoute payments have their own timeout inside the payment session. Can the timeout for regular payments also be moved into the regular payment session?